datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -452,6 +453,24 @@ private void validateRazEnablement(SdxClusterRequest sdxClusterRequest, Detailed
         }
     }
 
+    private void validateMediumDutyHaEnablement(SdxClusterRequest sdxClusterRequest, DetailedEnvironmentResponse environment) {
+        ValidationResultBuilder validationBuilder = new ValidationResultBuilder();
+        if (sdxClusterRequest.getClusterShape().equals(SdxClusterShape.MEDIUM_DUTY_HA)) {
```

Comment:
Swap this around, cluster shape could be null and thus null.equals throws a npe.